### ISSUE TITLE
您好，

### ISSUE THREAD

**Issue by niuxinhuai** [xmy0010/DemoForCSDN#2]
您好，请问下您有在WKWebView中使用自定义的UIMenuController吗？请问下如何关闭掉系统自带的复制，粘贴等选项？或者如何把自定义的item位置显示在最前边？

**Comment by xmy0010** [comment]
你好  我在代码中已经更新了。

**Comment by niuxinhuai** [comment]
很感谢你的回信。请问能否实现一下在WK中存在输入框的情况，点击输入框可以正常弹起，长按文字又能出现自定义的MeunItem。

**Comment by xmy0010** [comment]
是这样 你可以在特定的条件下改变 viewController是否可以取消第一响应，从而让其他控件响应（包括输入框）。另外还可以用UIMenuController出现和消失的系统通知。我已经更新到代码（在WK里面弹出点击‘添加笔记’），不知道你具体的需求是什么。  你可以看看 通过这俩点能不能达到你想要实现的效果

**Comment by niuxinhuai** [comment]
是的。正常只展示自定义WK通过第一响应可以实现只弹出自定义样式菜单。但是很多情况下用WK展示的视图底部都是会有评论区域的。我刚才试了您的方法在我的项目中。弹出键盘需要第一响应，当我再次点击WK弹出菜单栏的时候，是不能弹出自定义的会出现系统的copy等几个选项。

**Comment by niuxinhuai** [comment]
请问你方便吗？告诉我一下你的qq或者微信，我加您好友

**Comment by niuxinhuai** [comment]
测试可以发现，弹出键盘是需要dismissSelf为YES，才可以弹起的。但是当键盘手起我们再点击WK弹出菜单栏的时候，却是不可控的。因为系统会先走- (BOOL)canResignFirstResponder这个方法，再走menu的将要显示通知里边。这样没办法做到可控性

**Comment by xmy0010** [comment]
你可以尝试监听键盘弹出和收起的情况  试一试

**Comment by niuxinhuai** [comment]
试过了。他们这些通知的方法优先级低于系统的- (BOOL)canResignFirstResponder这个方法执行

**Comment by niuxinhuai** [comment]
或者换句话说，一旦- (BOOL)canResignFirstResponder这个方法在当前Controller生命周期内改为YES，这个方法将不再被调用

**Comment by xmy0010** [comment]
恩  这到确实是一个问题。  实在不行只能暂时不屏蔽系统几个按钮 保证项目功能实现正常。  我之前也在stack上面查过这个，没有做深入研究。 你可以看一下https://stackoverflow.com/questions/33243230/disable-entire-uimenucontroller-edit-menu-in-wkwebview

**Comment by niuxinhuai** [comment]
很感谢您，我会认真看下你给的文章

**Comment by xmy0010** [comment]
加了一个试图控制器 不知道是不是你想要的效果  可以参考一下

**Comment by xmy0010** [comment]
我的邮箱是18144200589@163.com  有问题可以一起探讨进步
